File: Android/app/src/main/java/com/khsm/app/data/entities/DisciplineResults.java
package com.khsm.app.data.entities;

import java.util.List;

public class DisciplineResults {
    public final Discipline discipline;
    public final List<Result> results;

    public DisciplineResults(Discipline discipline, List<Result> results) {
        this.discipline = discipline;
        this.results = results;
    }
}

File: Backend/Data/Entities/DisciplineResults.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Backend.Data.Entities
{
    public class DisciplineResults
    {
        [Required]
        public Discipline Discipline { get; set; }
        [Required]
        public IEnumerable<Result> Results { get; set; }
    }
}